### PR TITLE
Make sure setting faked properties doesn't throw an error (fixes #14)

### LINKED ISSUE #14: Faked properties throw error when assigned to twice
This should work the way you expect:

```js
var fake = leche.fake(someObject);

fake.foo = "bar";
fake.foo = "baz";
```

However, in Node.js v0.10.32, the second assignment throws "Attempted to assign to a readonly property".

Already verified this works fine in Chrome, so looks like an old V8 bug. I'm attempting to find a workaround.

### PR DESCRIPTION
**Pull request by nzakas**
Have to do a gross workaround to make this work in Node.js, but it works!

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nzakas has signed the CLA. Thanks for the pull request!

File: lib/leche.js [hunk 1]
```diff
@@ -199,14 +199,27 @@ module.exports = {
...
+							} else {
+								throw new Error('Unexpected use of property "' + propertyKey + '".');
+							}
 						},
 						set: function(value) {
 
-							// when set, change into a data property - removes the getter and setter
-							Object.defineProperty(fake, key, { value: value });
+							// normally we'd just switch to a data property, but this doesn't work in Node.js 0.10.32 :(
+							// Object.defineProperty(this, key, { value: value	});
```
Comment (j3tan) [inline lib/leche.js]: weird tab in comment

File: lib/leche.js [hunk 2]
```diff
@@ -199,14 +199,27 @@ module.exports = {
...
+							if (propertyIsSet) {
+								return propertyValue;
+							} else {
+								throw new Error('Unexpected use of property "' + propertyKey + '".');
+							}
 						},
 						set: function(value) {
 
-							// when set, change into a data property - removes the getter and setter
-							Object.defineProperty(fake, key, { value: value });
```
Comment (j3tan) [inline lib/leche.js]: curious, wouldn't this need to have writable: true?

File: lib/leche.js (hunk 1 above)
Comment (nzakas) [inline lib/leche.js]: Fixed.

File: lib/leche.js (hunk 2 above)
Comment (nzakas) [inline lib/leche.js]: Doh, you're totally right. That's a much simpler fix.

**Comment by j3tan** [comment]
+2

**Comment by nzakas** [comment]
Found another issue related to this, working on narrowing down the problem.

**Comment by j3tan** [comment]
+2, nice test case for the node issue

**Pull request merged**